src/posix/platform/include/openthread/openthread-system.h

```diff
@@ -311,6 +313,34 @@ bool otSysInfraIfIsRunning(void);
  */
 void otSysCliInitUsingDaemon(otInstance *aInstance);
 
+/**
+ * Sets whether to retrieve upstream DNS servers from "resolv.conf".
+ *
+ * @param[in] aEnabled  TRUE if enable retrieving upstream DNS servers from "resolv.conf", FALSE otherwise.
+ *
+ */
+void otSysUpstreamDnsServerSetResolvConfEnabled(bool aEnabled);
+
+/** Represents an upstream DNS server's address */
+typedef struct otSysUpstreamDnsServerAddress
+{
+    union
+    {
+        otIp6Address mIpv6;
+        otIp4Address mIpv4;
+    } mAddress;
+    bool mIsIpv4;
```

Comment:
This works. 

Alternate way can be to use a single `otIpv6Address` and then use "IPv4-mapped IPv6 address" format to specify an IPv4 address.

Reply:
BTW, OT provides helper function for conversion from/to this format: 
- `otIp4FromIp4MappedIp6Address()` and `otIp4ToIp4MappedIp6Address()`

Reply:
Thanks for the suggestion! Updated.